teuthology/orchestra/remote.py

```diff
@@ -53,8 +53,10 @@ def mkdtemp(self, suffix=None, parentdir=None):
 
         if suffix:
             args.append('--suffix=%s' % suffix)
-        if parentdir:
-            args.append('--tmpdir=%s' % parentdir)
+        if parentdir is None:
+            parentdir = misc.get_tmpdir()
+            self.sh(f"mkdir -p {parentdir}")
+        args.append('--tmpdir=%s' % parentdir)
```

Comment:
Here and later the fix is breaking the overall logic of the function. If we want to manipulate the default parent dir for remote temp files, it should definitely named correspondingly in config, like `remote_temp_parentdir` and read directly from the config, not from misc.

Given that by default remote_temp_parentdir is None the following code should be sufficient:
```
parentdir = parentdir or config.get('remote_temp_parentdir', None)
if parentdir:
     args.append('--tmpdir=%s' % parentdir)
```

Reply:
> If we want to manipulate the default parent dir for remote temp files, it should definitely named correspondingly in config, like remote_temp_parentdir and read directly from the config, not from misc.

Is that not what `tmp_path` in the misc function is fetching?

Reply:
yeah, the `tmp_path` is just not a good name, and also, since it is related to a _remote_ context it should not be in a misc, in general the `misc` looks like a garbage, and everything should be placed in right module instead of storing to misc.

Reply:
I would declare misc as deprecated at all.

Reply:
`tmpdir` maybe then? 

> since it is related to a remote context it should not be in a misc, in general the misc looks like a garbage, and everything should be placed in right module instead of storing to misc.

That's refactoring. I'm not going to blaze a trail to making the code more sensical in a bug fix.

Reply:
but you're not fixing, you are introducing new logic, the mktemp is exactly to use system mktemp, if you to fix test, go to test modules and pass parentdir argument.

Reply:
in other words this change is wasting API function, instead of fixing the real test logic, exactly test should manage the situation when it creates temp file in wrong place, this is why the parentdir argument exists.

Reply:
You're asking me to make widespread changes in the qa/ code to fix a problem created by Ubuntu 20.04 instead of fixing it one place in teuthology? I considered that but it's too much code to fix.

Reply:
First of all I ask not to not the base API complicated and difficult to maintain. But yes, if the test is written incorrectly for Ubuntu, fix it. If some one making a mess in client code, why not fix it, instead of doing something strange in the framework?

Reply:
is this only code that makes it failing:
```
        for (file_size, file_name) in zip(file_sizes, file_names):
            temp_file = self.mount_a.client_remote.mktemp(file_name)
            self.mount_a.run_shell(f"fallocate -l {file_size} {temp_file}")
            self.mount_a.run_shell(f'mv {temp_file} ./')
```

Reply:
also just imaging with this code you introduce several issues:
- hiding the real problem instead of resolving, reading the code it is not clear why the system mktemp cannot be used in general. And the code of the test is not written correctly and people do not understand why it works when it should not.
- this method will 100% fail if the /tmp/cephtest does not exists, why it must be created before the method called?
- user reading the doc for remote.mktemp does not understand it should have created /tmp/cephtest on remote host